staging/src/k8s.io/sample-apiserver/pkg/cmd/server/start.go

```diff
@@ -120,7 +120,9 @@ func (o *WardleServerOptions) Config() (*apiserver.Config, error) {
 		return nil, fmt.Errorf("error creating self-signed certificates: %v", err)
 	}
 
-	o.RecommendedOptions.Etcd.StorageConfig.Paging = utilfeature.DefaultFeatureGate.Enabled(features.APIListChunking)
+	if o.RecommendedOptions.Etcd != nil {
```

Comment:
I am not opposed to adding a nil check here. But it seems that func `NewRecommendedOptions` always return a non-nil etcd option?
https://github.com/kubernetes/kubernetes/blob/01f15d709226cd4b6e8791570c19597812db7af3/staging/src/k8s.io/apiserver/pkg/server/options/recommended.go#L65-L66

Reply:
@jingyih yeah, and the users are free to overwritting the etcd-option to nil after instantiating the whole option. we may import the sample-apiserver and make our further customization, is it?

Reply:
Got it. Thanks!